Wikipedia/View Controllers/Nearby/NearbyViewController.m

```diff
@@ -122,12 +113,50 @@ - (BOOL)prefersStatusBarHidden {
     return NO;
 }
 
-- (void)viewWillDisappear:(BOOL)animated {
-    [self.locationManager stopUpdatingLocation];
+- (void)viewWillAppear:(BOOL)animated {
+    [super viewWillAppear:animated];
+
+    self.skipStartUpdates = NO;
+    // Only begin updates if by the time the startUpdates selector below is performed
+    // self.skipStartUpdates is still NO - lets us add the nearby view controller
+    // somewhere and immediately cover it up without it starting to update locations
+    // and headings.
+    [self performSelector:@selector(startUpdates) withObject:nil afterDelay:0.5];
```

Comment:
Oh - I didn't notice the perform selector delay with delay here. We really shouldn't be doing this. 

I'm not exactly clear on what this is circumventing, but my hunch here is that the issue is because we are not putting this logic in viewDidAppear with the rest of the loading logic. 

Moving this there should would solve this issue. The whole point of having the methods with completion call backs is so we can control exactly when events happen. If we need to use a delay, that means we aren't using the completion blocks correctly. 

We should definitely resolve this before merging. This should totally be solvable without making any architecture changes.